### PR TITLE
Statemanager: Fix Cache Commit Bug

### PR DESCRIPTION
**Pull request by holgerd77**
This fixes a bug in both caches (account and storage) where the diff cache was wrongly updated if the changed account/storage key was not present in the lower CP diff cache.

### CODE REVIEW

**Review by jochem-brouwer — APPROVED** [review]
Before this PR:

`ts-node cli.ts --maxPeers 0 --executeBlocks 721873-721874`

Fail: invalid receipt trie.

After this PR: passes.

Also passes on `721830-721874`

**Comment by holgerd77** [comment]
Ok, in between update before I'll do a merge update here: all relevant tests passing 🎉, only 3 client tests failing (these base fee ones), seems something old, right?

**Review by acolytec3 — APPROVED** [review]
I'ma go ahead and very optimistically approve.  Hopefully we've licked it at this point!

File: packages/statemanager/test/ethersStateManager.spec.ts [hunk 1]
```diff
@@ -182,7 +182,7 @@ tape('runTx custom transaction test', async (t) => {
     // The `MockProvider` is not able to load JSON files dynamically in browser so skipped in browser tests
     t.end()
   } else {
-    const common = new Common({ chain: Chain.Mainnet })
+    const common = new Common({ chain: Chain.Mainnet, hardfork: Hardfork.London })
```
Comment (jochem-brouwer) [inline packages/statemanager/test/ethersStateManager.spec.ts:185]: Why do we set this to London?

Comment (holgerd77) [reply packages/statemanager/test/ethersStateManager.spec.ts:185]: Test was failing with the latest HF updates to Shanghai default, so I set the to something fixed (if I recall correctly it was something test setup related).

**Pull request merged**